gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-kafka/src/main/java/io/gravitee/plugin/endpoint/kafka/error/KafkaReceiverErrorTransformer.java

```diff
@@ -49,12 +50,17 @@ public static <K, V> Function<Flux<ConsumerRecord<K, V>>, Publisher<ConsumerReco
         return consumerRecordFlux -> {
             AtomicReference<Consumer<K, V>> consumerRef = new AtomicReference<>();
             Sinks.Many<ConsumerRecord<K, V>> kafkaErrorSink = Sinks.many().unicast().onBackpressureError();
+            Disposable kafkaDisconnectionMonitor = handleKafkaDisconnection(consumerRef, kafkaErrorSink);
             return consumerRecordFlux
                 .zipWith(storeConsumerReference(qosStrategy, consumerRef), (c, o) -> c)
                 .materialize()
                 .mergeWith(kafkaErrorSink.asFlux().materialize())
                 .<ConsumerRecord<K, V>>dematerialize()
-                .doOnSubscribe(subscription -> handleKafkaDisconnection(consumerRef, kafkaErrorSink));
+                .doAfterTerminate(() -> {
```

Comment:
```suggestion
                .doFinally(() -> {
```
doAfterTerminate only covers onError and onComplete, while doFinally covers also doOnDispose/doOnCancel.